components/List/ListSearchHeader.vue

```diff
@@ -58,29 +61,35 @@ export default {
         };
     },
     created() {
-        this.debounceFunc = debounce((value) => {
-            this.searchResult = value;
-            this.$emit('searchResult', value);
-        }, 500);
+        this.debouncedSearch = debounce(this.onSearch, 500);
     },
-    watch: {
-        searchResult() {
-            this.$emit('searchResult', this.searchResult);
-        },
+    beforeDestroy() {
+        delete this.debouncedSearch;
     },
     computed: {
+        searchBtnIconComponent() {
+            return this.isSearchBtnClicked
+                ? () => import('~/components/Icon/Window/IconClose')
+                : IconSearch;
```

Comment:
Why you use synchronous and asynchronous import on this property? use asynchronous.

Reply:
It's no matter if it is sync nor async component in computed